src/tree_model_filter.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013-2015, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib::object::Downcast;
+use glib::translate::*;
+use std::mem::transmute;
+
+use glib_ffi::{gboolean, gpointer};
+use {TreeIter, TreeModel, TreeModelFilter, TreePath};
+use ffi::{GtkTreeIter, GtkTreeModel};
+
+unsafe extern "C" fn trampoline(this: *mut GtkTreeModel, iter: *mut GtkTreeIter,
+                                f: gpointer) -> gboolean {
+    callback_guard!();
+    let f: &Box<Fn(&TreeModel, &TreeIter) -> bool> = transmute(f);
+    f(&TreeModel::from_glib_none(this).downcast_unchecked(), &from_glib_borrow(iter))
+    .to_glib()
+}
+
+unsafe extern "C" fn destroy_closure(ptr: gpointer) {
+    callback_guard!();
+    Box::<Box<Fn(&TreeModel, &TreeIter) -> bool + 'static>>::from_raw(ptr as *mut _);
+}
+
+fn into_raw<F>(func: F) -> gpointer
+    where F: Fn(&TreeModel, &TreeIter) -> bool + 'static {
+    let func: Box<Box<Fn(&TreeModel, &TreeIter) -> bool + 'static>> =
+        Box::new(Box::new(func));
+    Box::into_raw(func) as gpointer
+}
+
+/*unsafe extern "C" fn modify_trampoline<T>(this: *mut GtkTreeModel, iter: *mut GtkTreeIter,
+                                          value: *mut GValue, column: i32, f: gpointer)
+where T: IsA<TreeModel> {
+    callback_guard!();
+    assert!(column >= 0, "column cannot be less than 0");
+    let f: &Box<Fn(&T, &TreeIter, &Value, u32)> = transmute(f);
+    f(&TreeModel::from_glib_none(this).downcast_unchecked(), &from_glib_borrow(iter),
+      &Value(*value), column as u32).to_glib()
+}
+
+unsafe extern "C" fn modify_destroy_closure<T>(ptr: gpointer) {
+    callback_guard!();
+    Box::<Box<Fn(&T, &TreeIter, &Value, u32) + 'static>>::from_raw(ptr as *mut _);
+}
+
+fn modify_into_raw<F, T>(func: F) -> gpointer
+    where F: Fn(&T, &TreeIter, &Value, u32) + 'static {
+    let func: Box<Box<Fn(&T, &TreeIter, &Value, u32) + 'static>> =
+        Box::new(Box::new(func));
+    Box::into_raw(func) as gpointer
+}*/
+
+impl TreeModelFilter {
+    pub fn new(child_model: &TreeModel, root: &mut TreePath) -> TreeModelFilter {
```

Comment:
Converting `TreeModel` to `TreeModelFilter` is only reason to writing manual `new`.
`root: Option<&TreePath>` as it can be NULL and don't seems need `mut`